lib/web_ui/lib/src/engine/clipboard.dart

```diff
@@ -67,8 +71,8 @@ abstract class PasteFromClipboardStrategy {
 /// See: https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API
 class ClipboardAPICopyStrategy implements CopyToClipboardStrategy {
   @override
-  void setData(String text) {
-    html.window.navigator.clipboard
+  Future<void> setData(String text) async {
+    await html.window.navigator.clipboard
         .writeText(text)
         .catchError((error) => print('Could not copy text: $error'));
```

Comment:
Can we avoid the mixed use of `await` and `.catchError()`? Since we already made the function `async` and we are using `await`, let's also wrap it in `try-catch` instead of `.catchError()`.

Reply:
done.

Thanks I avoided using future methods.

Reply:
In my opinion though, if we use FutureMethods the code will be a lot cleaner since we can also clean the "Future" from the interface:
  bool setData(String text) {
    html.window.navigator.clipboard.writeText(text).then((_) {
      return true;
    }).catchError((e) {
      print('copy is not successful ${e.message}');
      return false;
    });
}

Reply:
As discussed offline, `setData()` still has to return a `Future` if it does any async work.

Also, to clarify, my concern was with mixing `await` with `catchErrors`. I'm not against `catchError`.

Reply:
Sure! thanks for the dartpad! :)